### PR TITLE
Add option to pass progress callback

### PR DESCRIPTION
**Pull request by dvahram**

### CODE REVIEW

**Review by daniel-warren — CHANGES_REQUESTED** [review]
Suggesting minor code-style changes to match the existing style, but otherwise I'm happy with the proposed changes 👍

File: src/IDRCloudClient.php [hunk 1]
```diff
@@ -246,7 +250,8 @@ public static function convert($opt) {
             $endpoint,
             $result,
             $opt[self::KEY_PARAMETERS],
-            (array_key_exists(self::KEY_CONVERSION_TIMEOUT, $opt)) ? $opt[self::KEY_CONVERSION_TIMEOUT] : false
+            (array_key_exists(self::KEY_CONVERSION_TIMEOUT, $opt)) ? $opt[self::KEY_CONVERSION_TIMEOUT] : false,
+            isset($opt[self::KEY_PROGRESS_CALLBACK]) ? $opt[self::KEY_PROGRESS_CALLBACK] : null
```
Comment (daniel-warren) [inline src/IDRCloudClient.php:254]:
```suggestion
            (array_key_exists(self::KEY_PROGRESS_CALLBACK, $opt)) ? $opt[self::KEY_PROGRESS_CALLBACK] : null
```
(Not sure how this suggestion didn't get included with the review, but adding now)

**Comment by dvahram** [comment]
@daniel-warren I've addressed your code style suggestions. Please take a look. Please let me know if there is anything else.

**Review by daniel-warren — APPROVED** [review]
Looks good!

**Review by josrepo — APPROVED** [review]

**Review by KieranFrance — APPROVED** [review]

**Pull request merged**